homeassistant/components/recorder/statistics.py

```diff
@@ -1597,17 +1639,59 @@ def _statistics_during_period_with_session(
         types,
     )
 
-    # Return statistics combined with metadata
-    if period not in ("day", "week", "month"):
-        return result
-
     if period == "day":
-        return _reduce_statistics_per_day(result, types)
+        result = _reduce_statistics_per_day(result, types)
 
     if period == "week":
-        return _reduce_statistics_per_week(result, types)
+        result = _reduce_statistics_per_week(result, types)
 
-    return _reduce_statistics_per_month(result, types)
+    if period == "month":
+        result = _reduce_statistics_per_month(result, types)
+
+    if "change" in _types:
```

Comment:
Maybe make the inner part its own function `_augment_result_with_change` ?

Reply:
Fixed in https://github.com/home-assistant/core/pull/92823